main.tf

```diff
@@ -14,84 +14,97 @@
  * limitations under the License.
  */
 
-/******************************************
-  Locals configuration
- *****************************************/
+resource "random_id" "random_hash_suffix" {
+  byte_length = 4
+}
+
+#--------#
+# Locals #
+#--------#
 locals {
-  project_id           = "${var.project_id}"
-  should_download      = "${var.download_forseti == "true" ? true : false}"
-  skip_sendgrid_config = "${var.sendgrid_api_key == ""}"
+  random_hash     = "${random_id.random_hash_suffix.hex}"
+  network_project = "${var.network_project != "" ? var.network_project : var.project_id}"
 
   services_list = [
     "admin.googleapis.com",
-    "cloudresourcemanager.googleapis.com",
+    "appengine.googleapis.com",
     "bigquery-json.googleapis.com",
     "cloudbilling.googleapis.com",
+    "cloudresourcemanager.googleapis.com",
     "sql-component.googleapis.com",
     "sqladmin.googleapis.com",
     "compute.googleapis.com",
-    "deploymentmanager.googleapis.com",
     "iam.googleapis.com",
+    "cloudtrace.googleapis.com",
+    "container.googleapis.com",
+    "servicemanagement.googleapis.com",
+    "logging.googleapis.com",
+    "cloudasset.googleapis.com",
+    "storage-api.googleapis.com",
   ]
-  /*COMMAND TO REPLACE SCRIPT FOR AUTOMATION*/
-  launch_first               = "sh ${path.module}/scripts/accScript.sh ${var.gsuite_admin_email} &&"
-  launch_command_main        = "cd forseti-security && python install/gcp_installer.py --no-cloudshell --service-account-key-file ${var.credentials_file_path} --gsuite-superadmin-email ${var.gsuite_admin_email}"
-  launch_command_gcs         = "${var.gcs_location != "" ? format("--gcs-location %s", var.gcs_location) : "--gcs-location \"\""}"
-  launch_command_cloudsql    = "${var.cloud_sql_region != "" ? format("--cloudsql-region %s", var.cloud_sql_region) : "--cloudsql-region \"\"" }"
-  launch_command_sendgrid    = "${var.sendgrid_api_key != "" ? format("--sendgrid-api-key %s", var.sendgrid_api_key) : "--skip-sendgrid-config" }"
-  launch_command_email_notif = "${var.notification_recipient_email != "" && !local.skip_sendgrid_config ? format("--notification-recipient-email %s", var.notification_recipient_email) : ""}"
-  launch_command_list        = "${compact(list(local.launch_first, local.launch_command_main, local.launch_command_sendgrid, local.launch_command_cloudsql, local.launch_command_email_notif, local.launch_command_gcs))}"
-  launch_command_fmt         = "${join(" ", local.launch_command_list)}"
 }
 
-/*******************************************
-  Activate services
- *******************************************/
-resource "google_project_service" "activate_services" {
-  count   = "${length(local.services_list)}"
-  project = "${local.project_id}"
-
-  service = "${element(local.services_list, count.index)}"
+#-------------------#
+# Activate services #
+#-------------------#
+resource "google_project_service" "main" {
```

Comment:
This should be dropped in favour of the submodule being added to project-factory.

Reply:
The mentioned submodule (https://github.com/terraform-google-modules/terraform-google-project-factory/pull/133) is currently a pull request against an unreleased version of the Project Factory. If we add this dependency we'll need to add a module referencing either a commit or branch against the Project Factory - this seems undesirable. Do you have another approach in mind for adding this dependency that I'm not seeing?

Reply:
@aaron-lane what's the advantage of using the module vs the `google_project_service` resource ? Is it just the ability to choose if we want to enable them or not using the `enable_apis` boolean ? In our Forseti case why would we disable the activation of APIs ?

Reply:
@adrienthebo no, I didn't realize it was still under review. I guess this is okay to leave for now.

@ocervell the idea is that we centralize the logic for enabling APIs on projects, rather than reproducing it across each module.